src/components/map.tsx

```diff
@@ -160,6 +165,14 @@ function useMapInstanceHandlerEffects(
         });
       }
 
+      if (onClick) {
+        google.maps.event.addListenerOnce(newMap, 'idle', () => {
+          google.maps.event.addListener(newMap, 'click', () => {
```

Comment:
This is missing some code to remove the event-listener on unmount. For this, we could store the returned [`google.maps.MapsEventListener`](https://developers.google.com/maps/documentation/javascript/reference/event#MapsEventListener) object and call it's `remove()` method.

Since there's going to be a lot of events, I think this should be extracted into a separate internal hook (for example `useMapEventHandlers()` that takes care of adding and removing event-handlers based on the props. Something similar has been done in [src/components/advanced-marker.tsx](https://github.com/visgl/react-google-maps/blob/main/src/components/advanced-marker.tsx).